### PR TITLE
[FEATURE] Active layer feature locator filter

### PR DESCRIPTION
**Pull request by nyalldawson**
Searches for matching attributes in any field from the current active layer.

The speed is actually reasonable... given that the search will be compiled most of the time, it'll automatically take advantage of any applicable indexes on the provider side. And even when it's slow... it's not hurting anything because it runs in the background :)

### CODE REVIEW

**Comment by nirvn** [comment]
Brilliant stuff Nyall. I'm assuming selecting a feature then pan/zoom canvas too?

**Comment by nyalldawson** [comment]
> I'm assuming selecting a feature then pan/zoom canvas too?

Yes - that's correct. I think we need to add a nice "flash feature" ability to the canvas. But that's another piece of work :)

File: src/app/locator/qgsinbuiltlocatorfilters.cpp [hunk 1]
```diff
@@ -163,3 +164,80 @@ void QgsActionLocatorFilter::searchActions( const QString &string, QWidget *pare
...
+  context.appendScopes( QgsExpressionContextUtils::globalProjectLayerScopes( layer ) );
+  dispExpression.prepare( &context );
+
+  Q_FOREACH ( const QgsField &field, layer->fields() )
+  {
+    if ( feedback->isCanceled() )
+      return;
+
+    QString exp = QStringLiteral( "%1 ILIKE '%%2%'" ).arg( QgsExpression::quotedColumnRef( field.name() ),
+                  string );
```
Comment (wonder-sk) [inline src/app/locator/qgsinbuiltlocatorfilters.cpp]: Possible SQL injection? :-)

**Review by wonder-sk — APPROVED** [review]
Nice! :+1: 

I hope at one point we will have a nice cache structure for this kind of full-text search... even though here it may use provider indices, for most columns it will be still slow sequential scan (multiple times)

**Comment by nirvn** [comment]
@nyalldawson , should it also open the identity panel? That would be useful.

**Comment by nirvn** [comment]
@nyalldawson , tested the PR, working great. Few thoughts:
* It'd be really helpful to have the identify results panel pop up (in addition to panning / zooming) -- it'd also effectively serve as a way to highlight the feature (via the identity result red glow)
* I feel the absence of a "loading" animation while searching across a large dataset isn't that great UX, it'd be worth adding a spinner (to differentiate searching vs nothing found state).

File: src/app/locator/qgsinbuiltlocatorfilters.cpp (hunk 1 above)
Comment (nyalldawson) [inline src/app/locator/qgsinbuiltlocatorfilters.cpp]: I think it's fine - at least, any fix belongs in the individual expression compiler's quotedValue methods, not here.

**Comment by nyalldawson** [comment]
> It'd be really helpful to have the identify results panel pop up (in addition to panning / zooming) -- it'd also effectively serve as a way to highlight the feature (via the identity result red glow)

This doesn't seem possible - at least, not without some large reworking of the identify tool/results dialog. I'll leave that to some possible future work.

**Comment by nyalldawson** [comment]
unrelated OSX test failure

**Pull request merged**